binary/expression.proto

```diff
@@ -18,6 +18,16 @@ message Expression {
         SwitchExpression switch_expression = 7;
         SingularOrList singular_or_list = 8;
         MultiOrList multi_or_list = 9;
+        Enum enum = 10;
+    }
+
+    message Enum {
+        oneof enum_kind {
+            string specified = 1;
+            Empty unspecified = 2;
+        }
+
+        message Empty {}
     }
```

Comment:
```suggestion
    message Enum {
        optional string value = 1;
    }
```
Maybe I'm misunderstanding protobof?

Reply:
You could do this but it makes thing more implicit. The way I've done it makes it clearer (albeit slightly larger on the wire). Optional is implicit in pb3 so doesn't have to be stated. The union also allows us to introduce more than two options later.

The other nice thing is that the "enum_kind" becomes something to switch off of in several languages.